drivers/gpio/Kconfig.mcp23x17

```diff
@@ -1,21 +1,39 @@
-# MCP23S17 GPIO configuration options
+# MCP23X17 GPIO configuration options
 
 # Copyright (c) 2020 Geanix ApS
+# Copyright (c) 2021 Prevas A/S
 # SPDX-License-Identifier: Apache-2.0
 
-menuconfig GPIO_MCP23S17
-	bool "MCP23S17 SPI-based GPIO chip"
-	depends on SPI
+menuconfig GPIO_MCP23X17
```

Comment:
Changing the Kconfig variable names needs to be done in the previous commit, so we don't have a commit that won't build the driver because the CMakeLists uses an undefined Kconfig symbol.  Probably all instances of the string `MCP23S17` should change to `MCP23X17` in that commit too.  That can be described as "change driver naming to reflect device family".  Note the other variables in this file also need to be renamed at that time, along with any references to them in the code (but no other changes).

The change of the description to include the I2C capability belongs in this commit where the code is updated to add the new bus interface.